flatland_viz/src/spawn_model_tool.cpp

```diff
@@ -173,34 +170,35 @@ void SpawnModelTool::deactivate() {
 }
 
 void SpawnModelTool::SpawnModelInFlatland() {
-  RCLCPP_INFO_STREAM(rclcpp::get_logger("SpawnModelTool"),  
+  RCLCPP_INFO_STREAM(rclcpp::get_logger("SpawnModelTool"),
       "SpawnModelTool::SpawnModelInFlatland name:" << model_name.toStdString());
-  flatland_msgs::msg::SpawnModel srv;
+  auto srv = std::make_shared<flatland_msgs::srv::SpawnModel::Request>();
 
   // model names can not have embeded period char
   model_name = model_name.replace(".", "_", Qt::CaseSensitive);
 
   // fill in the service request
-  srv.request.name = model_name.toStdString();
-  srv.request.ns = model_name.toStdString();
-  srv.request.yaml_path = path_to_model_file_.toStdString();
-  srv.request.pose.x = intersection[0];
-  srv.request.pose.y = intersection[1];
-  srv.request.pose.theta = initial_angle;
+  srv->name = model_name.toStdString();
+  srv->ns = model_name.toStdString();
+  srv->yaml_path = path_to_model_file_.toStdString();
+  srv->pose.x = intersection[0];
+  srv->pose.y = intersection[1];
+  srv->pose.theta = initial_angle;
 
-  client = nh.serviceClient<flatland_msgs::msg::SpawnModel>("spawn_model");
+  std::shared_ptr<rclcpp::Node> node = rclcpp::Node::make_shared("spawn_model_tool");  // TODO
```

Comment:
Not sure if it is a good way to create the node locally here or better as member variable?